libindex/fetcher.go

```diff
@@ -312,7 +312,7 @@ func (a *RemoteFetchArena) realizeLayer(ctx context.Context, l *claircore.Layer)
 	switch _, err := tarfs.New(fd); {
 	case errors.Is(err, nil):
 	case errors.Is(err, tarfs.ErrFormat):
-		fallthrough
+		return "", fmt.Errorf("tarfs error with layer %s: %w", l.Hash.String(), err)
```

Comment:
I think this should happen at the caller, so all returned errors get prefixed with the layer information